tests/FunctionsTest.php

```diff
@@ -206,6 +206,15 @@ public function testWithScope(): void
         $this->assertTrue($callbackInvoked);
     }
 
+    public function testWithScopeWithCallback(): void
```

Comment:
It doesn't make sense to pass the variable as reference to the lambda function if you then assign the return value of the callback to it from the outer scope. This test can be merged with the previous one instead and we can simply check the return value